lib/hammer_cli_foreman/associating_commands.rb

```diff
@@ -234,10 +234,48 @@ module ProvisioningTemplate
 
       class AddProvisioningTemplateCommand < HammerCLIForeman::AddAssociatedCommand
         associated_resource :provisioning_templates
-        desc _("Associate a provisioning template")
+        desc _("Associate provisioning templates")
+
+        option "--provisioning-template-ids", "PROVISIONING_TEMPLATE_IDS", _("List of provisioning template ids"),
+               format: HammerCLI::Options::Normalizers::List.new
+        option "--provisioning-templates", "PROVISIONING_TEMPLATE_NAMES", _("List of provisioning template names"),
+               format: HammerCLI::Options::Normalizers::List.new
+        option "--provisioning-template-search", "PROVISIONING_TEMPLATE_SEARCH", _("Provisioning template name regex to search, all matching templates will be associated")
+
+        def request_params
+          params = super
+          template_ids = params['operatingsystem']['provisioning_template_ids']
+          template_ids |= options['option_provisioning_template_ids'].map(&:to_s) if options['option_provisioning_template_ids']
+          if options['option_provisioning_templates']
+            options['option_provisioning_templates'].each do |provisioning_template_name|
+              template = HammerCLIForeman.collection_to_common_format(
+                associated_resource.action(:index).call(
+                  search: "name = \"#{provisioning_template_name}\"", per_page: 1000, page: 1))
+              raise ResolverError.new(_("%s not found.") % associated_resource.singular_name, associated_resource) if template.empty?
+
+              template_id = template.first['id']
+              template_ids << template_id.to_s
+            end
+          end
```

Comment:
We can avoid this repetitive (I'm quite sure we might need the same code in the future). Please, see my previous comment how.